.github/actions/repo-state/action.yml

```diff
@@ -0,0 +1,51 @@
+name: 'Repository state'
+description: 'Get the latest tag, the latest process-note, the number of commits since the last tag and the version numbers from the last tag.'
+outputs:
+  tag:
+    description: "Last tagged version"
+    value: ${{ steps.tag.outputs.value }}
+  note:
+    description: "Note on the last tag"
+    value: ${{ steps.note.outputs.value }}
+  commits:
+    description: "Number of commits since the last tag"
+    value: ${{ steps.commits.outputs.value }}
+  version_major:
+    description: "Major version number"
+    value: ${{ steps.version.outputs.major }}
+  version_minor:
+    description: "Minor version number"
+    value: ${{ steps.version.outputs.minor }}
+  version_revision:
+    description: "Revision number"
+    value: ${{ steps.version.outputs.revision }}
+runs:
+  using: "composite"
+  steps:
+    - uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+        ref: 'master'
```

Comment:
The workflow runs per default with the commit that was in `HEAD` when it is started, so `checkout` will always refer to this commit. This is bad for us, because we change some git-data or add another commit in the process, so we tell `checkout` not to use the context it got from the workflow but to work with the actual `HEAD` of `master`.

Reply:
I don't think I totally follow yet. Isn't this workflow just searching back through the history to find tags and notes? What git data is being changed in this workflow?

Reply:
This action does not add anything, it is read-only, that is correct.
The workflow `Release Bugfix` might add a new commit during `job: Check` and add a new release-tag to it if there was no commit after the last tag. If we get to `job: Publish` we use this action to get the tag that was created earlier. But by default we would neither see the commit nor the tag because they did not exist before the workflow was started. `ref: master` makes sure we can reach commits that were added during one of our workflows.

Reply:
Found another discussion that might explain this better than I can.
https://github.com/orgs/community/discussions/25985

They solve the problem slightly different, but it will help to understand the underlying problem.